data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -46,19 +46,26 @@ class SnapshotDelta(val graphs: MutableList[Snapshot],
 
   override def snapshotAtTime(instant: Instant): AttributeGraph = {
 
-    val closestGraph = graphs.reduce(returnClosestGraph(instant))
+    //Closest graph should be based on number of logs, not time.
+    // But for it to be feasible each graph should have a sequence ID as well
+    // This is possible, I think, but not implemented
+    val closestGraph = graphs.reduceLeft((snap1, snap2) =>
+      if (snap2.instant > instant) { 
+        snap1
+      } else {
+        snap2
+      })
```

Comment:
Kan vi ekstrahere denne reducing-a til closestGraph-funksjonen? Kan hende jeg misforstår intensjonen her tho

Reply:
```suggestion
    val closestGraph = getClosestGraph(graphs, instant) = 
```

```def returnClosestGraph(graph: SomeGraphType, instant: Instant): Snapshot =
graphs.reduceLeft((snap1, snap2) =>
      if (snap2.instant > instant) { 
        snap1
      } else {
        snap2
      })
```

Eller er det kanskje ikke så rett frem kanskje, idk

Reply:
jo den er ikke dum